source/fab/steps/preprocess.py

```diff
@@ -84,45 +85,41 @@ def process_artefact(self, fpath):
         Writes the output file to the output folder, with a lower case extension.
 
         """
-        with Timer() as timer:
-            # output_fpath = self.output_path(fpath)
-            output_fpath = input_to_output_fpath(
-                source_root=self._config.source_root,
-                workspace=self._config.project_workspace,
-                input_path=fpath).with_suffix(self.output_suffix)
-
-            # for dev speed, but this could become a good time saver with, e.g, hashes or something
-            if self._config.reuse_artefacts and output_fpath.exists():
-                log_or_dot(logger, f'Preprocessor skipping: {fpath}')
-                return output_fpath
-
-            if not output_fpath.parent.exists():
+        output_fpath = input_to_output_fpath(
+            source_root=self._config.source_root,
+            workspace=self._config.project_workspace,
+            input_path=fpath).with_suffix(self.output_suffix)
+
+        # already preprocessed?
+        if self._config.reuse_artefacts and output_fpath.exists():
```

Comment:
As noted, this logic is actually the wrong way round. The presence of an output file does not control the task, the presence of a (changed) input file controls the task. A change is not required at the moment but when incremental builds are implemented.